plugin/evm/gossip_mempool.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/avalanchego/network/p2p/gossip"
+
+	"github.com/ava-labs/coreth/core"
+	"github.com/ava-labs/coreth/core/txpool"
+	"github.com/ava-labs/coreth/core/types"
+)
+
+var (
+	_ gossip.Gossipable        = (*GossipEthTx)(nil)
+	_ gossip.Gossipable        = (*GossipAtomicTx)(nil)
+	_ gossip.Set[*GossipEthTx] = (*GossipEthTxPool)(nil)
+)
+
+type GossipAtomicTx struct {
+	Tx *Tx
+}
+
+func (tx *GossipAtomicTx) GetID() ids.ID {
+	return tx.Tx.ID()
+}
+
+func (tx *GossipAtomicTx) Marshal() ([]byte, error) {
+	return tx.Tx.SignedBytes(), nil
+}
+
+func (tx *GossipAtomicTx) Unmarshal(bytes []byte) error {
+	result := &Tx{}
+
+	_, err := Codec.Unmarshal(bytes, result)
+	if err != nil {
+		return err
+	}
+	unsignedBytes, err := Codec.Marshal(codecVersion, result.UnsignedAtomicTx)
```

Comment:
This should have the typeID
```suggestion
	unsignedBytes, err := Codec.Marshal(codecVersion, &result.UnsignedAtomicTx)
```